### PR TITLE
Allow --recommends , --no-recommends, --allow-vendor-change, --no-all…

### PR DESCRIPTION
**Pull request by gigi206**
### What does this PR do?
Allow to specify options --recommends , --no-recommends, --allow-vendor-change, --no-allow-vendor-change

These options must be in kwargs to be called by the state pkg.

### What issues does this PR fix or reference?
Fixes:

### Previous Behavior
Not possible to specify the --recommends option if --no--recommends is the default options by default.
Not possible to specify the --recommends options from the state pkg
Not possible to specify --allow-vendor-change and --no-allow-vendor-change

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [x] Docs
- [ ] Changelog - https://docs.saltstack.com/en/master/topics/development/changelog.html
- [ ] Tests written/updated

### Commits signed with GPG?
Yes

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

File: salt/modules/zypperpkg.py [hunk 1]
```diff
@@ -1428,7 +1410,6 @@ def install(
     skip_verify=False,
     version=None,
     ignore_repo_failure=False,
-    no_recommends=False,
```
Comment (Ch3LL) [inline salt/modules/zypperpkg.py:1431]: You cannot remove this as someone could possibly be calling these kwargs as positional args.

Also why not add `recommends=False` to the function, similar to `no_recommends`?

File: salt/modules/zypperpkg.py [hunk 2]
```diff
@@ -1717,9 +1704,7 @@ def upgrade(
     dryrun=False,
     dist_upgrade=False,
     fromrepo=None,
-    novendorchange=False,
```
Comment (Ch3LL) [inline salt/modules/zypperpkg.py:1720]: same here. you cannot remove kwargs since someone can be calling this function with positional args.

**Review by Ch3LL — CHANGES_REQUESTED** [review]
we will also need test coverage for this change and a changelog entry.

File: salt/modules/zypperpkg.py (hunk 1 above)
Comment (gigi206) [reply salt/modules/zypperpkg.py:1431]: Yes that's why I use `kwargs.get("no_recommends") is True `.

> Also why not add `recommends=False` to the function, similar to `no_recommends`?
I use kwargs but effectively I can add *recommends* to the function if your prefer (also same thing for *vendorchange*)

Comment (Ch3LL) [reply salt/modules/zypperpkg.py:1431]: > Yes that's why I use kwargs.get("no_recommends") is True 

This still would not cover the case where someone was calling the kwargs as positional args. Its not something they should be doing, but if we do not need to remove it I would rather keep the order to ensure that use case also works on upgrade.

> I use kwargs but effectively I can add recommends to the function if your prefer (also same thing for vendorchange)

I prefer to not use kwargs unless there is a valid use case when we need to use it.

**Comment by dwoz** [comment]
Closing this due to inactivity. Anyone should feel free to re-open it if they want to see it through to the end in one release cycle.

**Pull request closed without merging**